File: app/src/main/java/com/promiseland/ks/base/utils/FieldHelper.java
package com.promiseland.ks.base.utils;

import java.util.Collection;
import java.util.List;

public final class FieldHelper {
    public static final boolean isEmpty(Collection<?> collection) {
        return collection != null ? collection.isEmpty() : true;
    }

    public static final boolean isNotEmpty(Collection<?> collection) {
        return !(collection != null ? collection.isEmpty() : true);
    }

    public static final boolean isEmpty(CharSequence str) {
        return str == null || str.length() == 0;
    }

//    public static final boolean isEmpty(Image image) {
//        return isEmpty((CharSequence) image != null ? image.getUrl() : null);
//    }

//    public static final boolean equals(CharSequence a, CharSequence b) {
//        if (a == b) {
//            return true;
//        }
//        if (a == null || b == null || a.length() != b.length()) {
//            return false;
//        }
//        if ((a instanceof String) && (b instanceof String)) {
//            return Intrinsics.areEqual(a, b);
//        }
//        boolean z;
//        Iterator it = new IntRange(0, a.length() - 1).iterator();
//        while (it.hasNext()) {
//            int it2 = ((IntIterator) it).nextInt();
//            if (a.charAt(it2) != b.charAt(it2)) {
//                z = true;
//                continue;
//            } else {
//                z = false;
//                continue;
//            }
//            if (z) {
//                z = false;
//                break;
//            }
//        }
//        z = true;
//        return z;
//    }

    public static final boolean isLargerZero(Integer value) {
        return value != null && value.intValue() > 0;
    }

    public static final double getPrimitive(Double value) {
        return value != null ? value.doubleValue() : 0.0d;
    }

    public static final int getPrimitive(Integer value) {
        return value != null ? value.intValue() : 0;
    }

//    public static final String getPrimitive(Image image) {
//        return image != null ? image.getUrl() : null;
//    }

    public static final float getPrimitive(Float value) {
        return value != null ? value.floatValue() : 0.0f;
    }

    public static final boolean hasPosition(List<?> list, int position) {
        if (position < 0) {
            return false;
        }
        return position < (list != null ? list.size() : 0);
    }

    public static final int getListSize(List<?> list) {
        return list != null ? list.size() : 0;
    }

//    public static final <T, R> List<R> mapList(List<? extends T> list, Function1<? super T, ? extends R> mapFunction) {
//        Intrinsics.checkParameterIsNotNull(mapFunction, "mapFunction");
//        if (list == null) {
//            return null;
//        }
//        ArrayList mappedList = new ArrayList(list.size());
//        for (Object item$iv : list) {
//            mappedList.add(mapFunction.invoke(item$iv));
//        }
//        Collection collection = mappedList;
//        return mappedList;
//    }

    public static final boolean isLastPosition(List<?> list, int position) {
        if (list != null) {
            if ((!((Collection) list).isEmpty() ? 1 : null) != null && position == list.size() - 1) {
                return true;
            }
        }
        return false;
    }

//    public static final boolean containsKeys(Map<?, ?> map, Object... keys) {
//        Intrinsics.checkParameterIsNotNull(keys, "keys");
//        if (map == null) {
//            return false;
//        }
//        Object[] $receiver$iv = keys;
//        for (Object element$iv : $receiver$iv) {
//            if (!map.containsKey(element$iv)) {
//                return false;
//            }
//        }
//        return true;
//    }
}

File: app/src/main/java/com/promiseland/ks/base/utils/FieldHelper.kt
package com.promiseland.ks.base.utils

import kotlin.jvm.internal.Intrinsics

/**
 * Created by Administrator on 2017/11/14.
 */
object FieldHelper {
    fun isEmpty(collection: Collection<*>?): Boolean {
        return collection?.isEmpty() ?: true
    }

    fun isNotEmpty(collection: Collection<*>?): Boolean {
        return !(collection?.isEmpty() ?: true)
    }

    fun isEmpty(str: CharSequence?): Boolean {
        return str == null || str.isEmpty()
    }

    fun equals(a: CharSequence?, b: CharSequence?): Boolean {
        if (a === b) {
            return true
        }
        if (a == null || b == null || a.length != b.length) {
            return false
        }
        if (a is String && b is String) {
            return Intrinsics.areEqual(a, b)
        }
        val it = IntRange(0, a.length - 1).iterator()
        while (it.hasNext()) {
            val index = it.nextInt()
            if (a[index] != b[index]) {
                return false
            }
        }
        return true
    }

    fun isLargerZero(value: Int?): Boolean {
        return value != null && value.toInt() > 0
    }

    fun getPrimitive(value: Double?): Double {
        return value ?: 0.0
    }

    fun getPrimitive(value: Int?): Int {
        return value ?: 0
    }

    fun getPrimitive(value: Float?): Float {
        return value ?: 0.0f
    }

    fun hasPosition(list: List<*>?, position: Int): Boolean {
        return if (position < 0) {
            false
        } else position < list?.size ?: 0
    }

    fun getListSize(list: List<*>?): Int {
        return list?.size ?: 0
    }

    fun <T, R> mapList(list: List<T>?, mapFunction: Function1<T, R>): List<R>? {
        Intrinsics.checkParameterIsNotNull(mapFunction, "mapFunction")
        if (list == null) {
            return null
        }
        val mappedList = mutableListOf<R>()
        list.mapTo(mappedList) { mapFunction.invoke(it) }
        return mappedList
    }

    fun isLastPosition(list: List<*>?, position: Int): Boolean {
        if (list != null) {
            if ((if (!(list as Collection<*>).isEmpty()) 1 else null) != null && position == list.size - 1) {
                return true
            }
        }
        return false
    }

    fun containsKeys(map: Map<*, *>?, vararg keys: Any): Boolean {
        Intrinsics.checkParameterIsNotNull(keys, "keys")
        if (map == null) {
            return false
        }
        return keys.any { map.containsKey(it) }
    }

}